packages/aws-cdk-lib/aws-dynamodb/lib/table.ts

```diff
@@ -551,6 +558,12 @@ export abstract class TableBase extends Resource implements ITable, iam.IResourc
    */
   public abstract resourcePolicy?: iam.PolicyDocument;
 
+  /**
+   * Resource policy to assign to table.
+   * @attribute
+   */
+  public abstract streamResourcePolicy?: iam.PolicyDocument;
```

Comment:
Related to this field, we should probably update the [`grantStream`](https://github.com/aws/aws-cdk/blob/a5c6167534cf2780b24486772b06e510c88b98a1/packages/aws-cdk-lib/aws-dynamodb/lib/table.ts#L604) and the [`grantStreamRead`](https://github.com/aws/aws-cdk/blob/a5c6167534cf2780b24486772b06e510c88b98a1/packages/aws-cdk-lib/aws-dynamodb/lib/table.ts#L658) methods.

And make this field `protected` to limit the exposure as there is no other constructs need to consume this at them moment.

Let me know what you think.